grab-site: 2.2.0 -> 2.2.2; pythonPackages.ludios_wpull: 3.0.7 -> 3.0.9

ivan opened the pull request:
###### Motivation for this change

This updates grab-site, broken at runtime since 9974c90a1597a7b2b009e798374615899d57db7d when `python3Packages.sqlalchemy` was upgraded to 1.4 in nixpkgs.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

ivan committed:
grab-site: 2.2.0 -> 2.2.2; pythonPackages.ludios_wpull: 3.0.7 -> 3.0.9

Use SQLAlchemy 1.3 because ludios_wpull is incompatible with SQLAlchemy 1.4.

Use Python 3.8 instead of Python 3.7, as namedlist 1.8 fixed compatibility
with Python 3.8. We cannot use Python 3.9 until
https://github.com/ArchiveTeam/ludios_wpull/issues/20 is fixed.

L-as reviewed (approved):
~~LGTM~~

L-as reviewed (changes requested):
See my latest comment.

ivan committed:
python3Packages.ludios_wpull: use updated URLs

https://github.com/ludios/wpull redirects to
https://github.com/ArchiveTeam/ludios_wpull

SuperSandro2000 commented on pkgs/development/python-modules/ludios_wpull/default.nix:
```diff
@@ -15,15 +15,15 @@
-  version = "3.0.7";
+  version = "3.0.9";
 
   disabled = (!isPy3k) || (pythonAtLeast "3.8");
```
```suggestion
  disabled = pythonAtLeast "3.8";
```
The project does not support python2 anymore https://github.com/ArchiveTeam/ludios_wpull/blob/master/setup.py#L81

ivan replied:
Thanks. I removed the `isPy3k` and bumped the disabled version to 3.9+, where ludios_wpull is incompatible.

SuperSandro2000 commented on pkgs/top-level/python-packages.nix:
```diff
@@ -8744,6 +8744,8 @@ in {
 
   sqlalchemy = callPackage ../development/python-modules/sqlalchemy { };
 
+  sqlalchemy_1_3 = callPackage ../development/python-modules/sqlalchemy/1.3.nix { };
```
This is not allowed. Please move this into grab-site.

ivan replied:
Thanks, fixed and back to matching the other packages that require SQLAlchemy 1.3.

SuperSandro2000 commented on pkgs/tools/backup/grab-site/default.nix:
```diff
@@ -3,25 +3,26 @@ let
   postPatch = ''
     substituteInPlace setup.py \
-      --replace '"wpull @ https://github.com/ArchiveTeam/ludios_wpull/tarball/master#egg=wpull-3.0.7"' '"wpull"'
+      --replace '"wpull @ https://github.com/ArchiveTeam/ludios_wpull/tarball/master#egg=wpull-3.0.9"' '"wpull"'
```
```suggestion
      --replace '"wpull @ https://github.com/ArchiveTeam/ludios_wpull/tarball/master#egg=wpull-${ludios_wpull.version}"' '"wpull"'
```

ivan replied:
Thanks, fixed.

SuperSandro2000 commented on pkgs/tools/backup/grab-site/default.nix:
```diff
@@ -3,25 +3,26 @@ let
   python = python37.override {
```
Do we really need to use python37?

ivan replied:
It looks like I do not. grab-site seems to work fine with Python 3.8, so I have changed it to use `python38`. Python 3.9 does not work because of https://github.com/ArchiveTeam/ludios_wpull/issues/20.

ivan commented on pkgs/tools/backup/grab-site/default.nix:
```diff
@@ -4,24 +4,32 @@ let
       tornado = super.tornado_4;
+
+      sqlalchemy = super.sqlalchemy.overridePythonAttrs (oldAttrs: rec {
+        version = "1.3.24";
```
`sqlalchemy_1_3` was not accepted in Sandro's later review, so I am marking this resolved.
